### ISSUE TITLE
ultralytics/yolo/engine/model.py Override option: 'save' is not used. Can't see results....

### ISSUE THREAD

**Issue by killuhwhale** [ultralytics/ultralytics#1743]
TL;DR
Line 223 should just be removed from ultralytics/yolo/engine/model.py

Repro steps:
1. Clone repo
2. Run model with overrides['save'] = True
3. Predict with model.

Expected:
 - Results saved to /home/<user>/ws_p38/project_name/runs/detect/predict8

Observed:
Nothing, no results saved. (Although, successful prediction occurred)

The problem:
I downloaded and set up Yolov8.

I wanted to see the bounding box from the results of object detection. 

I set overrides as follows:
```
model = YOLO(weights)
model.overrides = {
    'save': True,
    'save_conf': True,
    'verbose': False,
}

result = model(source_img)  # Calling with only source, no kwargs
```

The options save_crop and verbose work as expected. However, 'save' does not.
Line #223 from engines/models.py does a dumb useless thing and leaves the user unable to view the results using cfg or overrides attr.
223: overrides['save'] = kwargs.get('save', False)  # not save files by default

When callings the model, its __call__ method calls YOLO.predict(source, stream, kwargs) BUT kwargs are empty.
The quickstart guide shows how to use and configure a model, we can use cfg files or overrides attr. 

It seems like the default behavior should be described in the DEFAULT_CFG_PATH = ROOT / 'yolo/cfg/default.yaml'

What is the point of setting model.overrides AND needing to pass in the overrides to calling the model as shown below? 

```
model.overrides = {
    'save': True,
    'save_conf': True,
    'verbose': False,
}
result = model(source=source_img, **overrides)   # add overrides in order to pass them as kwargs to .predict()
```

Line 223 should just be removed.

**Comment by AyushExel** [comment]
The line doesn't do a dumb useless thing. It prevents saving files to your disk unless you want it.
The correct usage is this:
```
mode.predict(source, save=True)
```
You don't need to define model.overrides instead just pass all the required args in .predict function. Please refer the official docs on usage - https://docs.ultralytics.com/modes/predict/

**Comment by killuhwhale** [comment]
> AyushExel

I see now, thanks for the reply!

After a quick read of Configuration and Adv. Customization :
https://docs.ultralytics.com/usage/cfg/ AND
https://docs.ultralytics.com/usage/engine/

it lead me to believe the recommended way to configure was to use overrides or create a config file.

The link you gave doesn't mention how to use the options, it only describes the sources param and how to use results.
It was a bit annoying to figure out.... maybe it can be made a bit more clear earlier on? 



I found 1 mention of it here...
https://docs.ultralytics.com/usage/python/#predict
```
results = model.predict(source="folder", show=True) # Display preds. Accepts all YOLO predict arguments
```

It would be nice if it was mentioned in the quickstart or more explicitly mentioned. Currently, it's mentioned in an in-line comment.

**Comment by AyushExel** [comment]
Yes agreed. Thanks for the feedback. I have opened a PR to address this and should be resolved later today.

**Comment by Tejaswini-Jayashanker** [comment]
Namaste,
I did try all of the options mentioned above.
I tried "--save" and "--save_crop", but it doesn't work.  It is saving all images, the only difference is that we will have BBox around the class mentioned.
I want to save images that only have classes.

Can anyone help with a detailed explanation?

**Comment by pderrenger** [comment]
@Tejaswini-Jayashanker namaste! To save only the images with detected classes, you can use the `save` option combined with the `save_conf` option. By setting `save=True` and `save_conf=True`, YOLOv8 will save only the images with detected classes and their associated bounding boxes. This way, you can filter out the images with no detected classes. You can use these options when calling the `predict` function, or alternatively, you can set these options within the YOLOv8 configuration file. I hope this helps!
